### PR TITLE
Plumb UiBroker compatibility into pipes-shell

### PR DESCRIPTION
**Pull request by sjmiles**
... and restore some "classic" sharing/ingestion solutions to enable apps and demos (these are easily removed when the time comes, I believe they are not in the way).

Fwiw, apps that use this plumbing will be in other PRs.

### CODE REVIEW

File: src/runtime/ui-slot-composer.ts [hunk 1]
```diff
@@ -227,6 +236,22 @@ export class UiSlotComposer {
     }
   }
 
+  sendEvent(particleId, eventlet) {
+    // TODO(sjmiles): `arc` is bound to `this` in arc.ts specifically for use here, simplify
+    const arc = this['arc'];
+    if (arc) {
+      const particle = arc.activeRecipe.particles.find(
```
[the remark this answers is not in the document]
Comment (sjmiles) [reply src/runtime/ui-slot-composer.ts]: Great, thank you very much.

[the remark this answers is not in the document]
Comment (sjmiles) [reply src/runtime/ui-slot-composer.ts]: Do you have a way I get the PEC, as in line 250, without `this['arc']`?

[the remark this answers is not in the document]
Comment (mmandlis) [reply src/runtime/ui-slot-composer.ts]: a SlotConsumer has an `arc` data member

File: shells/pipes-shell/source/ingestion-arc.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+// TODO(sjmiles): "live context" tools below included so "classic" ingestion demos
+// continue to function. These methods are easily removed when the demos are no
+// longer important.
+
+import {logFactory} from '../../../build/runtime/log-factory.js';
```
[the remark this answers is not in the document]
Comment (sjmiles) [reply shells/pipes-shell/source/ingestion-arc.js]: See previous comment.

File: particles/Pipes/Ingestion.arcs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+schema IncomingEntity
```
[the remark this answers is not in the document]
Comment (sjmiles) [reply particles/Pipes/Ingestion.arcs]: Unless I messed up, all my "ingestion" stuff is disjoint from yours. `ingestion-arc` loads my version of the Arc for use with `Pipes/*`. Your material in `PipeApps` is untouched.

Presumably at some point my sharing solution isn't needed, and we just delete the non-Maria stuff.

**Pull request merged**